### ISSUE TITLE
Can't build with ASAN on Mac

### ISSUE THREAD

**Issue by Gerold103** [tarantool/tarantool#4237]
MacOSX Mojave 10.14.4
```
cmake . -DENABLE_ASAN=ON
make -j
```
My compiler is clang 10.0.1. Both cmake and make finished successfully. Then I start and terminate tarantool with Ctrl+D, but do not see any output. Then I tried to dereference a null pointer, I created a mem leak, but ASAN is silent.

Then I tried this example:
```C
#include <stdlib.h>

int main() {
	char *x = (char*)malloc(10 * sizeof(char*));
	free(x);
	return x[5];
}
```
In a file `test.c`.
```
clang -fsanitize=address -O1 -fno-omit-frame-pointer -g test.c
```
And now ASAN works perfectly. It means, that something is wrong with Tarantool's cmake files.

**Comment by Gerold103** [comment]
Appeared, that ASAN works, but no always. For example, it does not detect memory leaks, and some crashes, but does others. I added this to the source code:
```Diff
@@ -681,6 +681,8 @@ break_loop(struct trigger *, void *)
 int
 main(int argc, char **argv)
 {
+       volatile char *p = (char *) malloc(300);
+       (void) p;
        /* set locale to make iswXXXX function work */
        if (setlocale(LC_CTYPE, "C.UTF-8") == NULL &&
            setlocale(LC_CTYPE, "en_US.UTF-8") == NULL &&
```
And ASAN does not detect it. On the other hand it somehow detected use-after-free. Looks, like it works on half.

**Comment by avtikhon** [comment]
The issue still exists, checked the same manual change, I've tried it on version
2.5.0-2-g098324556

**Comment by locker** [comment]
ASAN works fine with OS X, e.g. see #7259.

As for the fact that apparent leaks aren't detected (see https://github.com/tarantool/tarantool/issues/4237#issuecomment-494724844), this happens, because we suppress **all** malloc leaks in our ASAN config. I filed an issue to fix this, see #7326.
